libraries/Adapters/Microsoft.Bot.Builder.Adapters.Facebook/FacebookHelper.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Copyright (c) Microsoft Corporation.All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.Bot.Schema;
+
+namespace Microsoft.Bot.Builder.Adapters.Facebook
+{
+    public static class FacebookHelper
+    {
+        /// <summary>
+        /// Converts an Activity object to a Facebook messenger outbound message ready for the API.
+        /// </summary>
+        /// <param name="activity">The activity to be converted to Facebook message.</param>
+        /// <returns>The resulting message.</returns>
+        public static FacebookMessage ActivityToFacebook(Activity activity)
+        {
+            var facebookMessage = new FacebookMessage(activity.Conversation.Id, new Message(), "RESPONSE");
+
+            facebookMessage.Message.Text = activity.Text;
+
+            // map these fields to their appropriate place
+            if (activity.ChannelData != null)
+            {
+                facebookMessage.MessagingType = (activity.ChannelData as dynamic).messaging_type != null ? (activity.ChannelData as dynamic).messaging_type : null;
```

Comment:
This can be simplified as `facebookMessage.MessagingType = (activity.ChannelData as dynamic).messaging_type ?? null;`
The same for the other fields.

Reply:
how does that differ from simply assigning `(activity.ChannelData as dynamic).messaging_type` to it?

Reply:
What you're doing is, if messaging_type is not null, assign messaging_type , if it's null, assign null.
My suggestion (Resharper) is the same. Assign messaging_type unless it's null, in that case, assign null.